### PR TITLE
expand stream structure for loans pallet

### PR DESCRIPTION
**Pull request by rjman-ljm**

### CODE REVIEW

File: pallets/streaming/src/types.rs [hunk 1]
```diff
@@ -14,6 +14,8 @@ pub enum StreamStatus {
     Completed,
```
Comment (GopherJ) [inline pallets/streaming/src/types.rs:14]: should put inside Ongoing

Comment (rjman-ljm) [reply pallets/streaming/src/types.rs:14]: Why do it, I don't understand

Comment (GopherJ) [reply pallets/streaming/src/types.rs:14]: how about Ongoing { asCollateral: false|true }

**Comment by rjman-ljm** [comment]
@GopherJ I've pushed some changes

**Comment by rjman-ljm** [comment]
1. current implementation
2. struct Ongoing { asCollateral: false | true } &&  struct Completed { cancelled: false | true }

Which one is better

**Comment by rjman-ljm** [comment]
@GopherJ I've pushed some changes

File: pallets/streaming/src/types.rs (hunk 1 above)
Comment (rjman-ljm) [reply pallets/streaming/src/types.rs]: okay

File: pallets/streaming/src/types.rs [hunk 2]
```diff
@@ -70,9 +75,13 @@ impl<T: Config> Stream<T> {
             recipient,
             start_time,
             end_time,
-            status: StreamStatus::Ongoing,
+            status: StreamStatus::Ongoing {
+                as_collateral: false,
+            },
+            cancellable: true,
```
Comment (i5possible) [inline pallets/streaming/src/types.rs:81]: Should we add it to the create method and set the default true if not provided?

Comment (rjman-ljm) [reply pallets/streaming/src/types.rs:81]: If we support creating a non-cancellable stream, we need to add a new field to create(), and we must pass in field

**Review by i5possible — APPROVED** [review]
LGTM

**Comment by i5possible** [comment]
error: linking with `cc` failed: exit status: 1

Could you compile it in local?

File: pallets/streaming/src/types.rs (hunk 2 above)
Comment (rjman-ljm) [reply pallets/streaming/src/types.rs]: Already added

**Comment by rjman-ljm** [comment]
@GopherJ The PR for fixing CI was merged, please do a review for these changes

**Review by GopherJ — APPROVED** [review]

**Pull request merged**